spec/request/isolation_segments_spec.rb

```diff
@@ -214,14 +281,119 @@
-                'spaces' => { 'href' => "/v2/spaces?q=isolation_segment_guid:#{models[1].guid}" }
+                'spaces' => { 'href' => "/v3/isolation_segments/#{models[1].guid}/relationships/spaces" },
+                'organizations' => { 'href' => "/v3/isolation_segments/#{models[1].guid}/relationships/organizations" },
+              }
+            },
+            {
+              'guid'        =>  models[2].guid.to_s,
+              'name'        =>  models[2].name.to_s,
+              'created_at'  =>  iso8601,
+              'updated_at'  =>  nil,
+              'links'       =>  {
+                'self' => { 'href' => "/v3/isolation_segments/#{models[2].guid}" },
+                'spaces' => { 'href' => "/v3/isolation_segments/#{models[2].guid}/relationships/spaces" },
+                'organizations' => { 'href' => "/v3/isolation_segments/#{models[2].guid}/relationships/organizations" },
               }
             },
           ]
         }
 
         expect(parsed_response).to be_a_response_like(expected_response)
       end
+
+      it 'filters by isolation segment names' do
+        get "/v3/isolation_segments?names=#{models[2].name}%2C#{models[4].name}", nil, user_header
+
+        expected_pagination = {
+          'total_results' =>  2,
+          'total_pages'   =>  1,
+          'first'         =>  { 'href' => "/v3/isolation_segments?names=#{models[2].name}%2C#{models[4].name}&page=1&per_page=50" },
+          'last'          =>  { 'href' => "/v3/isolation_segments?names=#{models[2].name}%2C#{models[4].name}&page=1&per_page=50" },
+          'next'          =>  nil,
+          'previous'      =>  nil
+        }
+
+        parsed_response = MultiJson.load(last_response.body)
+
+        expect(last_response.status).to eq(200)
+        expect(parsed_response['resources'].map { |r| r['name'] }).to eq([models[2].name, models[4].name])
+        expect(parsed_response['pagination']).to eq(expected_pagination)
+      end
+
+      it 'filters by isolation segment guids' do
+        get "/v3/isolation_segments?guids=#{models[3].guid}%2C#{models[5].guid}", nil, user_header
+
+        expected_pagination = {
+          'total_results' =>  2,
+          'total_pages'   =>  1,
+          'first'         =>  { 'href' => "/v3/isolation_segments?guids=#{models[3].guid}%2C#{models[5].guid}&page=1&per_page=50" },
+          'last'          =>  { 'href' => "/v3/isolation_segments?guids=#{models[3].guid}%2C#{models[5].guid}&page=1&per_page=50" },
+          'next'          =>  nil,
+          'previous'      =>  nil
+        }
+
+        parsed_response = MultiJson.load(last_response.body)
+
+        expect(last_response.status).to eq(200)
+        expect(parsed_response['resources'].map { |r| r['name'] }).to eq([models[3].name, models[5].name])
+        expect(parsed_response['pagination']).to eq(expected_pagination)
+      end
+
+      context 'when the user is not an admin' do
```

Comment:
This would also be more appropriate as a controller spec. Generally, we want the request specs to only cover super happy path cases since they are relatively slow.

Correct me if I'm wrong @utako